Development/nmos/jwt_validator_impl.cpp

```diff
@@ -0,0 +1,466 @@
+                                {
+                                    if (!decoded_token.has_payload_claim(x_nmos_scope_claim_)) { return false; }
+                                    const auto x_nmos_scope_claim = decoded_token.get_payload_claim(x_nmos_scope_claim_).to_json();
+
+                                    if (!x_nmos_scope_claim.is_null())
+                                    {
+                                        auto accessible = [&x_nmos_scope_claim, req, &path](const std::string& access_right)
+                                        {
+                                            if (x_nmos_scope_claim.contains(access_right))
+                                            {
+                                                auto accessible_paths = jwt::basic_claim<nlohmann_traits>(x_nmos_scope_claim.at(access_right)).as_array();
+                                                for (auto& accessible_path : accessible_paths)
+                                                {
+                                                    // construct path regex for regex comparison
+
+                                                    auto acc_path = accessible_path.get<std::string>();
+                                                    // replace any '*' => '.*'
+                                                    boost::replace_all(acc_path, "*", ".*");
+                                                    const bst::regex path_regex(acc_path);
+                                                    if (bst::regex_match(path, path_regex))
+                                                    {
+                                                        return true;
+                                                    }
+                                                }
+                                            }
+                                            return false;
+                                        };
+
+                                        // write accessible
+                                        if ((web::http::methods::POST == req.method())
+                                            || (web::http::methods::PUT == req.method())
+                                            || (web::http::methods::PATCH == req.method())
+                                            || (web::http::methods::DEL == req.method()))
+                                        {
+                                            return accessible("write");
+                                        }
+
+                                        // read accessible
+                                        if ((web::http::methods::OPTIONS == req.method())
+                                            || (web::http::methods::GET == req.method())
+                                            || (web::http::methods::HEAD == req.method()))
+                                        {
```

Comment:
Hi @garethsb, please have a look of it, there must be a better way to verify the private claim!

Reply:
This LGTM but some unit testing of the `accessible` function would be desirable